### ISSUE TITLE
ZeRO Optimizer and Mistral Checkpoint

### ISSUE THREAD

**Issue by IsNoobgrammer** [pytorch-tpu/transformers#58]
### Feature request

Is There any possibility to add Gradient Checkpointing for mistral and 
ability to use ZeRO optimizer or 1bit ADAM optimizer .. 

### Motivation

.

### Your contribution

.
